src/include/OpenImageIO/memory.h

```diff
@@ -0,0 +1,80 @@
+// Copyright 2008-present Contributors to the OpenImageIO project.
+// SPDX-License-Identifier: BSD-3-Clause
+// https://github.com/OpenImageIO/oiio
+
+
+/////////////////////////////////////////////////////////////////////////////
+/// \file
+///
+/// Provides helper function to compute the memory footprint of common
+/// containers.
+///
+/////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#define OPENIMAGEIO_MEMORY_H
+
+OIIO_NAMESPACE_BEGIN
+
```

Comment:
Good suggestion, I didn't know where to put that !
I've first implemented a templated version but removed it because of the "if constexpr is a c++17 extension" warnings.

Reply:
We have C++17 minimum.

Reply:
In master, I mean.

I think if there's part of this that simply must be backported to 2.5, then we can use enable_if tricks rather than `if constexpr` for those.